plugins/util/portmap/chain.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2017 CNI authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+
+	"github.com/coreos/go-iptables/iptables"
+)
+
+type chain struct {
+	name        string
+	entryRule   []string // the rule that enters this chain
+	entryChains []string // the chains to add the entry rule
+}
+
+func (c *chain) setup(ipt *iptables.IPTables, rules [][]string) error {
+	// Create the chain and add its rules
+	if err := ipt.NewChain("nat", c.name); err != nil {
+		return err
+	}
+	for _, rule := range rules {
+		if err := ipt.AppendUnique("nat", c.name, rule...); err != nil {
```

Comment:
@bboreham looks like it will wait indefinitely if you don't give it a specific number of seconds.  IIRC there was some older version that people still ship that doesn't allow a seconds option, so nobody uses it by default, just infinite wait.